File: cross-application-feature-development-management/NotepadPlusPlusFileManagementCommandSwitcher.cs
namespace cross_application_feature_development_management
{
    public class NotepadPlusPlusFileManagementCommandSwitcher(
        ProcessManager processManager,
        CloseProcessManagement closeProcessManagement
        )
    {
        private string GetCommand()
        {
            var command = CommandLineArgs.GetByKey("--command");
            return command;
        }

        private bool IsOpen()
        {
            return GetCommand() == "open";
        }

        private bool IsClose()
        {
            return GetCommand() == "close";
        }

        public void Run()
        {
            if (IsOpen())
            {
                processManager.Run();
            }
            else if (IsClose())
            {
                closeProcessManagement.Run();
            }
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/NotepadPlusPlusFileManagement/NotepadPlusPlusFileManagementCommandSwitcher.fs
namespace cafdemwimu.console.Applications.NotepadPlusPlusFileManagement

open cafdemwimu.console

type NotepadPlusPlusFileManagementCommandSwitcher
    (
        processManager: ProcessManager,
        closeProcessManagement: CloseProcessManagement
    ) =
    member private _.GetCommand() =
        let command = CommandLineArgs.GetByKey("--command")
        command

    member private this.IsOpen() = this.GetCommand() = "open"

    member private this.IsClose() = this.GetCommand() = "close"

    member this.Run() =
        if this.IsOpen() then
            processManager.Run()
        elif this.IsClose() then
            closeProcessManagement.Run()
